backend/src/common/pipes/validation.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { PipeTransform, Injectable, ArgumentMetadata, BadRequestException } from '@nestjs/common';
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { AppErrorTypeEnum } from '../error/AppErrorTypeEnum';
+import { AppError } from '../error/AppError';
+import { Logger } from '../../modules/logger/logger.service'
+
+@Injectable()
+export class ValidationPipe implements PipeTransform<any> {
+  async transform(value, { metatype }: ArgumentMetadata) {
+    if (!metatype || !this.toValidate(metatype)) {
```

Comment:
`!this.toValidate(metatype)` sitas type checkas nereikalingas, nes metatype tipas visada bus
`    readonly metatype?: new (...args) => any | undefined;`.
Bet siaip reiketu vengt tokiu savadariu type checku, nebent type yra nezinomas runtime :D Siai problemai sprest ir yra skirtas typescript